osgp/protocol-adapter-dlms/osgp-protocol-adapter-dlms/src/test/java/org/opensmartgridplatform/adapter/protocol/dlms/application/threads/RecoverKeyProcessTest.java

```diff
@@ -86,6 +90,44 @@ void testWhenDeviceNotFoundThenException() throws OsgpException {
     verify(this.secretManagementService, never()).activateNewKeys(any(), any(), any());
   }
 
+  @Test
+  void testWhenAlreadyKeyProcessRunning() throws OsgpException, IOException {
+
+    // GIVEN
+    when(this.domainHelperService.findDlmsDevice(DEVICE_IDENTIFICATION, IP_ADDRESS))
+        .thenReturn(DEVICE);
+    when(this.deviceKeyProcessingService.isProcessing(DEVICE_IDENTIFICATION)).thenReturn(true);
+
+    // WHEN
+    this.recoverKeyProcess.run();
```

Comment:
I was worried about the behavior of the test suite when this method would be executed.
The `deviceKeyProcessingService.isProcessing` returning `true` causes an exception that is handled by scheduling another run of the `RecoverKeyProcess` with a `TimerTask`.
To get an idea of how this would play out, I executed this test case from my IDE, and instead of some hanging JVM, I got:
```
org.mockito.exceptions.verification.TooManyActualInvocations: 
domainHelperService.findDlmsDevice(
    "E000123456789",
    "1.1.1.1"
);
Wanted 1 time:
-> at org.opensmartgridplatform.adapter.protocol.dlms.application.services.DomainHelperService.findDlmsDevice(DomainHelperService.java:82)
But was 3 times:
-> at org.opensmartgridplatform.adapter.protocol.dlms.application.threads.RecoverKeyProcess.findDevice(RecoverKeyProcess.java:145)
-> at org.opensmartgridplatform.adapter.protocol.dlms.application.threads.RecoverKeyProcess.findDevice(RecoverKeyProcess.java:145)
-> at org.opensmartgridplatform.adapter.protocol.dlms.application.threads.RecoverKeyProcess.findDevice(RecoverKeyProcess.java:145)
	at org.opensmartgridplatform.adapter.protocol.dlms.application.services.DomainHelperService.findDlmsDevice(DomainHelperService.java:82)
	at org.opensmartgridplatform.adapter.protocol.dlms.application.threads.RecoverKeyProcessTest.testWhenAlreadyKeyProcessRunning(RecoverKeyProcessTest.java:106)
```

If a repeat this, or run more tests it apparently undeterministically fails with above exception, or reports to succeed. We should not have tests with this uncontrolled behavior.

Reply:
This multithreaded process will not be tested in JUnit tests using mocks. This situation will be tested in Cucumber tests